File: Week 02/id_063/C++/LeetCode_94_063.cpp

#include <stdio.h>
#include <vector>
using namespace std;


 struct TreeNode {
     int val;
     TreeNode *left;
     TreeNode *right;
     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 };

class Solution {
public:

    void dfs(vector<int>& result, TreeNode* p_cur_node) {
        if (p_cur_node == nullptr) {
            return;
        }

        dfs(result, p_cur_node->left);
        result.push_back(p_cur_node->val);
        dfs(result, p_cur_node->right);
    }

    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> v;
        dfs(v, root);

        return v;
    }
};

File: Week 02/id_063/Java/LeetCode_94_063.java
/*
思路
简单中序遍历，没什么特殊算法
*/

public class Solution {
    private List<Integer> dfs(List<Integer> result, TreeNode curNode) {
        if (curNode == null) {
            return result;
        }

        dfs(result, curNode.left).add(curNode.val);
        return dfs(result, curNode.right);
    }


    public List<Integer> inorderTraversal(TreeNode root) {
        return dfs(new LinkedList<>(), root);
    }
}